scripts/release.sh

```diff
@@ -1,10 +1,126 @@
-echo 'twine upload dist/* -r https://upload.pypi.org/legacy/ -u kubernetes'
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
+# Workflow
+# 1. [master branch] update existing snapshot (include API change for a new alpha/beta/GA
+# release)
+#   - add a new snapshot or reuse the existing snapshot, the latter means either
+#     API change happened in a k8s patch release, or we want to include some new
+#     python / python-base change in the release note
+#   - API change w/ release notes
+#   - master change w/ release notes
+#   - submodule change w/ release notes
+# 2. [master branch] create new snapshot (include API change for a new alpha release)
+#   - add a new snapshot or reuse the existing snapshot, the latter means either
+#     API change happened in a k8s patch release, or we want to include some new
+#     python / python-base change in the release note
+#   - API change w/ release notes
+#   - master change w/ release notes
+#   - submodule change w/ release notes
+# 3. [release branch] create a new release
+#   - pull master
+#     - it's possible that master has new changes after the latest snaphost,
+#       update CHANGELOG accordingly
+#   - for generated file, resolve conflict by committing the master version
+#   - abort if a snapshot doesn't exists
+#   - generate client change, abort if API change is detected
+#   - CHANGELOG: latest snapshot becomes the release, create a new snapshot
+#     section that reflect the master branch state
+#   - README: add the release to README
+#   - an extra PR to update CHANGELOG and README in master in sync with this new
+#     release
+#
+# Difference between 1&2: API change release notes
+#
+# TODO(roycaihw):
+# - add user input validation
+# - add function input validaiton (release/version strings start with 'v' or not)
+# - automatically send a PR; provide useful links for review
+#   - master branch diff: https://github.com/kubernetes-client/python/compare/commit1..commit2
+#   - python base diff: https://github.com/kubernetes-client/python-base/compare/commit1..commit2
+#   - Kubernetes changelog, e.g. https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.18.md
+# - add debug log
+# - add a sentence about "changes since {last release}". In most cases our
+#   releases should be sequential. This script (the workflow above) is based on
+#   this assumption, and we should make the release note clear about that.
```

Comment:
@yliaog This (and the Workflow above) is the sequential releases I mentioned in https://github.com/kubernetes-client/python/pull/1425#discussion_r618788294.

Reply:
ok. let me know once the PR is ready for review.

Reply:
I generated a 19.0.0-snapshot client using this script: https://github.com/kubernetes-client/python/pull/1547

I think we can submit the current progress and develop on top of it. Please take a look.